app/models/miq_worker/runner.rb

```diff
@@ -370,13 +383,12 @@ def heartbeat
     messages = worker_monitor_drb.worker_heartbeat(@worker.pid, @worker.class.name, @worker.queue_name)
     @last_hb = now
     messages.each { |msg, *args| process_message(msg, *args) }
-    do_heartbeat_work
-  rescue DRb::DRbError => err
-    do_exit("Error heartbeating to MiqServer because #{err.class.name}: #{err.message}", 1)
-  rescue SystemExit, SignalException
-    raise
-  rescue Exception => err
-    do_exit("Error heartbeating because #{err.class.name}: #{err.message}\n#{err.backtrace.join('\n')}", 1)
+  end
+
+  def heartbeat_to_file
+    require 'fileutils'
```

Comment:
My two cents with this:

Just include it at the top of the file.  This costs us around a megabyte to include it, and there is a good chance it is already being included by some other gem anyway.

If we do it this way, it means we are making this `require "fileutils"` call on every `heartbeat_to_file` call, and I would personally want to make this as lightweight of an operation as possible (though, I think subsequent requires are pretty good at being quick, but still).